index 4444e21..28160f7 100644
--- a/src/editline.c
+++ b/src/editline.c
@@ -6,6 +6,7 @@
 -----------------------------------------------------------------------------*/
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "common.h"
 #include "term.h"
@@ -881,6 +882,10 @@ static char* edit_line( ic_env_t* env, const char* prompt_text )
   // process keys
   code_t c;          // current key code
   while(true) {    
+    if (!isatty(0)) {
+      return NULL;
+    }
+
     // read a character
     term_flush(env->term);
     if (env->hint_delay <= 0 || sbuf_len(eb.hint) == 0) {
diff --git a/src/term.c b/src/term.c
index c453046..1a2c641 100644
--- a/src/term.c
+++ b/src/term.c
@@ -23,9 +23,6 @@
 #include <unistd.h>
 #include <errno.h>
 #include <sys/ioctl.h>
-#if defined(__linux__)
-#include <linux/kd.h>
-#endif
 #endif
 
 #define IC_CSI      "\x1B["
